### PR TITLE
Student Orgs

### LINKED ISSUE #216: Student org database
Thoughts on either importing the existing student organization database hosted on Oleville (not sure if that is up-to-date), or creating a mirrored copy of [St. Olaf's student org list](https://www.stolaf.edu/orgs/list/index.cfm) and descriptions?

I am thinking if Student Orgs are incorporated into the app, we might be able to get away with creating another scraper (akin to the dictionary scraper) that collects the data and notifies on change.

### PR DESCRIPTION
**Pull request by drewvolz**
This introduces an AlphabetListView to display the many Student Organizations that live at [St. Olaf's CheckImHere](https://stolaf.checkimhere.com/organizations).

The main list displays organization:
* name
* categories

The detail view when a row is tapped on will display:
* name
* categories
* meeting time (if any)
* meeting location (if any)
* contact info (if any)
* description (if any)

The detail view has to make a query of its own -- not all of the info for an org is located in the main query for the orgs. We adjust for this by presenting a LoadingView in case the request goes for longer than planned, and by presenting the info we did receive for the first request immediately into the view. Tricky!

I have decided not to list the number of members in this set of info. I feel as if it might discourage someone from joining an organization they feel is too (large/small). That is up for debate and I am willing to hear opinions on it.

@hawkrives thank you for helping clean up the two views, for providing types for everything, for the two-part rendering magic as described in the detail view, and finally for making this code production-quality.

@elijahverdoorn reported he is talking to IT about getting access to the checkimhere API so that we may possibly gain access to info that is not available on the public API without a token of sorts.

tl;dr we have student org data to display

Closes #216

| List | Details |
|:---:|:---:|
list | details

### CODE REVIEW

**Comment by hawkrives** [comment]
We had some minor issues during a rebase, but we've got it sorted now.

**Review by hawkrives — CHANGES_REQUESTED** [review]
Looks good! A few changes I'd like to request:

- [X] Change the blue dot icon to `c.infoBlue`?
- [X] Tweak the space between the title and the first `<Section/>` – it's larger than others
- [X] Switch to fetchJson?
- [X] Fix alignment of the text with separators – they're off a bit. You can take a peek at [fancy-menus](https://github.com/StoDevX/AAO-React-Native/pull/452/files#diff-856c08e76db88779b9c45c200f149a91R14) – I did something similar for the stars in the gutter.

**Comment by drewvolz** [comment]
I think those four commits satisfy your requests.

**Review by elijahverdoorn — APPROVED** [review]
This looks good and I like the concept. I got the notification that the API that you're using was up a while ago, was meaning to inform everyone, seems that you found it already. One less thing! 👍

**Review by hawkrives — APPROVED** [review]

**Comment by hawkrives** [comment]
<table><td></td></table>
